cmd/crank/beta/validate/validate_test.go

```diff
@@ -0,0 +1,1198 @@
+								Kind:   "TestClaim",
+								Plural: "testclaims",
+								Categories: []string{
+									"claim",
+								},
+							},
+							Scope: "Namespaced",
+							Versions: []extv1.CustomResourceDefinitionVersion{
+								{
+									Name:    "v1alpha1",
+									Served:  true,
+									Storage: false,
+									Subresources: &extv1.CustomResourceSubresources{
+										Status: &extv1.CustomResourceSubresourceStatus{},
+									},
+									AdditionalPrinterColumns: []extv1.CustomResourceColumnDefinition{
+										{
+											Name:     "SYNCED",
+											Type:     "string",
+											JSONPath: ".status.conditions[?(@.type=='Synced')].status",
+										},
+										{
+											Name:     "READY",
+											Type:     "string",
+											JSONPath: ".status.conditions[?(@.type=='Ready')].status",
+										},
+										{
+											Name:     "CONNECTION-SECRET",
+											Type:     "string",
+											JSONPath: ".spec.writeConnectionSecretToRef.name",
+										},
+										{
+											Name:     "AGE",
+											Type:     "date",
+											JSONPath: ".metadata.creationTimestamp",
+										},
+									},
+
+									Schema: &extv1.CustomResourceValidation{
+										OpenAPIV3Schema: &extv1.JSONSchemaProps{
+											Type: "object",
+											Required: []string{
+												"spec",
+											},
+											Properties: map[string]extv1.JSONSchemaProps{
+												"apiVersion": {
+													Type: "string",
+												},
+												"kind": {
+													Type: "string",
+												},
+												"metadata": {
+													Type: "object",
+													Properties: map[string]extv1.JSONSchemaProps{
+														"name": {
+															Type:      "string",
+															MaxLength: ptr.To[int64](63),
+														},
+													},
+												},
+												"spec": {
+													Type: "object",
+													Properties: map[string]extv1.JSONSchemaProps{
+														"replicas": {
+															Type: "integer",
+														},
+														"compositeDeletePolicy": {
+															Type: "string",
+															Enum: []extv1.JSON{
+																{Raw: []byte(`"Background"`)},
+																{Raw: []byte(`"Foreground"`)},
+															},
+														},
+														"compositionRef": {
+															Type: "object",
+															Required: []string{
+																"name",
+															},
+															Properties: map[string]extv1.JSONSchemaProps{
+																"name": {
+																	Type: "string",
+																},
+															},
+														},
+														"compositionRevisionRef": {
+															Type: "object",
+															Required: []string{
+																"name",
+															},
+															Properties: map[string]extv1.JSONSchemaProps{
+																"name": {
+																	Type: "string",
+																},
+															},
+														},
+														"compositionRevisionSelector": {
```

Comment:
phew these test objects are gnarly! 🐻  have you found any tips to make it easier to navigate table driven test cases when they get this big? 📏

Reply:
the secret is Github Copilot 🙈

Reply:
Would using JSON strings to represent objects instead of `map[string]any` help at all?